### PR TITLE
feat: Support Portal

### PR DESCRIPTION
**Pull request by AfshanKhan**
Screenshot 2020-06-25 at 3 15 26 PM
Screenshot 2020-06-25 at 3 15 39 PM

### CODE REVIEW

File: erpnext/www/support/index.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				`tabWeb Page View` AS t2 
+			ON t1.route = t2.path 
+			WHERE t1.published = 1
+			GROUP BY route 
+			ORDER BY count DESC
+			LIMIT 6;
+			""", as_dict=True)
+
+def get_latest_articles():
+	return frappe.db.sql(
```
Comment (scmmishra) [inline erpnext/www/support/index.py]: Use `frappe.get_list` (ORM) here instead of SQL query

File: erpnext/www/support/index.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	context.greeting_title = setting.greeting_title
+	context.greeting_subtitle = setting.greeting_subtitle
+	
+	# Support content
+	context.favorite_article_list=[]
+	context.help_article_list=[]
+	context.category_list = frappe.get_all("Help Category", fields="name")
+	favorite_articles = get_favorite_articles_by_page_view()
+	if not favorite_articles or len(favorite_articles) < 3:
```
Comment (scmmishra) [inline erpnext/www/support/index.py]:
```suggestion
	if not favorite_articles:
```

If you get a list, show it anyway, or add latest articles next to it. So if you get 2 favorite articles, add 4 recent ones

**Review by scmmishra — APPROVED** [review]

**Review by scmmishra — CHANGES_REQUESTED** [review]
Minor changes mentioned above

**Comment by AfshanKhan** [comment]
> Minor changes mentioned above

@scmmishra  Changes have been pushed

File: erpnext/www/support/index.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	favorite_articles = get_favorite_articles_by_page_view()
+	if len(favorite_articles) < 6:
+		title_list = []
+		if favorite_articles:
+			for article in favorite_articles:
+				title_list.append(article.title)
+		latest_articles = frappe.get_all("Help Article", 
+		fields=["title", "content", "route", "category"], 
+		filters={"title": ['not in', tuple(title_list)], "published": 1}, 
+		order_by="creation desc", limit=(6-len(favorite_articles)))
```
Comment (scmmishra) [inline erpnext/www/support/index.py]:
```suggestion
		latest_articles = frappe.get_all("Help Article", 
			fields=["title", "content", "route", "category"], 
			filters={"title": ['not in', tuple(title_list)], "published": 1}, 
			order_by="creation desc", limit=(6-len(favorite_articles)))
```

File: erpnext/www/support/index.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	latest_articles = None
+	favorite_articles = get_favorite_articles_by_page_view()
+	if len(favorite_articles) < 6:
+		title_list = []
+		if favorite_articles:
+			for article in favorite_articles:
+				title_list.append(article.title)
+		latest_articles = frappe.get_all("Help Article", 
+		fields=["title", "content", "route", "category"], 
+		filters={"title": ['not in', tuple(title_list)], "published": 1},
```
Comment (scmmishra) [inline erpnext/www/support/index.py]: Use name instead of title

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Review by scmmishra — DISMISSED** [review]

**Comment by surajshetty3416** [comment]
@AfshanKhan Please add screenshots & description. Also, always try to write a good fitting PR title.

https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

**Review by Anurag810 — APPROVED** [review]

**Pull request merged**